Add a clang-tidy GH action

jmillan opened the pull request:
A github action that runs on every PR and executes clang-tidy on the changed files.

It also adds suggestions to the offending code which can be directly committed, although such suggestions are not formatted following .clang-format, hence `make format` may need to be run locally and push the changes.

jmillan committed:
WIP: add clang-tidy on CI

jmillan committed:
Revert "WIP: add clang-tidy on CI"

This reverts commit f6a20c94e5aca8bd9143a08fc1ecc09a9f657a85.

jmillan committed:
add clang-tidy-review.yaml workflow

jmillan committed:
update action version

ibc commented:
Please some description about what is going on here

jmillan commented:
It's a mere test, I'm trying to add a clang-tidy GH action. I'm affraid it's going to take me some trials. Please unsubscribe from this draft. I'll set it as PR when ready.

jmillan committed:
WIP

jmillan committed:
install invoke

jmillan committed:
remove condition

jmillan committed:
cont

jmillan committed:
cont

jmillan committed:
cont

jmillan committed:
add build_dir with the path to compile_commands.json

jmillan committed:
add build_dir with the path to compile_commands.json

ibc commented on .github/workflows/clang-tidy-review.yaml:
```diff
@@ -0,0 +1,34 @@
+          clang_tidy_version: '17'
+          include: '*.hpp,*.cpp'
+
+      # Uploads an artefact containing clang_fixes.json
```
"artifact" right?

jmillan replied:
I'm just trying to make it work for the moment. I'll probably give up on it, because it's a nightmare. I'll change this suggestions but this is far from being ready.

ibc commented on .github/workflows/clang-tidy-review.yaml:
```diff
@@ -0,0 +1,34 @@
+      - uses: ZedThree/clang-tidy-review/upload@v0.22.1
+        id: upload-review
+
+      # If there are any comments, fail the check
```
```suggestion
      # If there are any comments, fail the check.
```

ibc commented on .github/workflows/clang-tidy-review.yaml:
```diff
@@ -0,0 +1,34 @@
+          clang_tidy_version: '17'
+          include: '*.hpp,*.cpp'
+
+      # Uploads an artefact containing clang_fixes.json
```
```suggestion
      # Uploads an artifact containing clang_fixes.json.
```

ibc commented on .github/workflows/clang-tidy-review.yaml:
```diff
@@ -0,0 +1,34 @@
+name: clang-tidy-review
+
+# You can be more specific, but it currently only works on pull requests
```
```suggestion
# You can be more specific, but it currently only works on pull requests.
```

jmillan committed:
cont

jmillan committed:
cont

jmillan committed:
make @ibc happy

jmillan committed:
specify mediasoup source files

jmillan committed:
fix include param

jmillan committed:
fix string concatenation

jmillan committed:
fix include attr

jmillan committed:
copy complie_commands.json

jmillan committed:
cont

jmillan committed:
foo

jmillan committed:
cont

jmillan commented:
clang-tidy review says "All clean, LGTM! :+1:"

jmillan committed:
generate compile_commands.json via install_commands

jmillan committed:
does not find compile_commands.json

jmillan commented:
clang-tidy review says "All clean, LGTM! :+1:"

jmillan commented:
clang-tidy review says "All clean, LGTM! :+1:"

jmillan commented:
clang-tidy review says "All clean, LGTM! :+1:"

jmillan committed:
cont

jmillan committed:
cont

jmillan commented:
clang-tidy review says "All clean, LGTM! :+1:"

jmillan committed:
cont

jmillan commented:
clang-tidy review says "All clean, LGTM! :+1:"

ibc commented on .github/workflows/clang-tidy-review.yaml:
```diff
@@ -0,0 +1,33 @@
+name: clang-tidy-review
```
Please rename to `mediasoup-clang-tidy` (also remove the "-review" suffix) and make that change in this line, in the file name and please add it to the README with other CI status/links.

jmillan committed:
Rename action

jmillan committed:
run the action if the actions .yaml file changes

jmillan committed:
use clang-tidy version 17

jmillan commented:
clang-tidy review says "All clean, LGTM! :+1:"

jmillan commented:
clang-tidy review says "All clean, LGTM! :+1:"

jmillan committed:
install libclang-rt-21-dev

To avoid clang-tidy complain about some header not being found which is required
by sanitizer.

See more here https://github.com/ZedThree/clang-tidy-review/issues/157#issuecomment-3557516297

jmillan commented:
clang-tidy review says "All clean, LGTM! :+1:"

jmillan commented:
clang-tidy review says "All clean, LGTM! :+1:"

jmillan committed:
cont

jmillan commented:
clang-tidy review says "All clean, LGTM! :+1:"

jmillan commented:
clang-tidy review says "All clean, LGTM! :+1:"

jmillan commented:
clang-tidy review says "All clean, LGTM! :+1:"

jmillan commented:
clang-tidy review says "All clean, LGTM! :+1:"

jmillan commented:
clang-tidy review says "All clean, LGTM! :+1:"

jmillan commented:
clang-tidy review says "All clean, LGTM! :+1:"

jmillan commented:
clang-tidy review says "All clean, LGTM! :+1:"

jmillan commented:
clang-tidy review says "All clean, LGTM! :+1:"

jmillan commented:
clang-tidy review says "All clean, LGTM! :+1:"

jmillan committed:
use './build' directory as build_dir

jmillan committed:
fix typo

jmillan commented:
clang-tidy review says "All clean, LGTM! :+1:"

jmillan commented:
clang-tidy review says "All clean, LGTM! :+1:"

jmillan committed:
cont

jmillan commented:
clang-tidy review says "All clean, LGTM! :+1:"

jmillan committed:
avoid using github.workspace variable

jmillan commented:
clang-tidy review says "All clean, LGTM! :+1:"

jmillan commented:
clang-tidy review says "All clean, LGTM! :+1:"

jmillan committed:
adjust build dir

jmillan commented:
clang-tidy review says "All clean, LGTM! :+1:"

jmillan commented:
clang-tidy review says "All clean, LGTM! :+1:"

jmillan commented:
clang-tidy review says "All clean, LGTM! :+1:"

jmillan commented:
clang-tidy review says "All clean, LGTM! :+1:"

jmillan commented:
clang-tidy review says "All clean, LGTM! :+1:"

jmillan commented:
clang-tidy review says "All clean, LGTM! :+1:"

jmillan commented:
clang-tidy review says "All clean, LGTM! :+1:"

jmillan commented:
clang-tidy review says "All clean, LGTM! :+1:"

jmillan commented:
clang-tidy review says "All clean, LGTM! :+1:"

jmillan commented on worker/src/RTC/SvcConsumer.cpp:
```diff
@@ -88,7 +88,7 @@ namespace RTC
 		// libsrtp bug:
 		// https://github.com/versatica/mediasoup/issues/1437
-		const uint16_t initialOutputSeq =
+		uint16_t initialOutputSeq =
```
warning: variable 'initialOutputSeq' of type 'uint16_t' (aka 'unsigned short') can be declared 'const' [[misc-const-correctness](https://clang.llvm.org/extra/clang-tidy/checks/misc/const-correctness.html)]

```suggestion
		uint16_t const initialOutputSeq =
```

jmillan commented on worker/src/RTC/SvcConsumer.cpp:
```diff
@@ -694,7 +694,7 @@ namespace RTC
 
 		// Whether this is the first packet after re-sync.
-		const bool isSyncPacket = this->syncRequired;
+		bool isSyncPacket = this->syncRequired;
```
warning: variable 'isSyncPacket' of type 'bool' can be declared 'const' [[misc-const-correctness](https://clang.llvm.org/extra/clang-tidy/checks/misc/const-correctness.html)]

```suggestion
		bool const isSyncPacket = this->syncRequired;
```

jmillan reviewed (commented):
clang-tidy made some suggestions

jmillan committed:
cont

jmillan committed:
normalize compile commands to match clang-tidy filter

jmillan committed:
Put github workflows back

jmillan committed:
put .github workflows back

ibc commented on worker/scripts/normalize-compile-commands.mjs:
This script must be added to the arrays in `npm-scripts.mjs` to be prettier and ESLint aware.

jmillan replied:
`worker/scripts`  folder is already added there.

ibc replied:
True.

jmillan committed:
rename script

jmillan committed:
Add comment

jmillan committed:
Remove duplicated comment

ibc commented on .github/workflows/mediasoup-worker-clang-tidy.yaml:
```diff
@@ -0,0 +1,50 @@
+          clang_tidy_version: '21'
+          # TMP until this bug is fixes https://github.com/ZedThree/clang-tidy-review/issues/157#issuecomment-3552341821.
+          extra_arguments: '--allow-no-checks'
+          # pr: 1661
```
What does this mean exactly?

ibc replied:
I mean the `# pr: 1661` comment.

jmillan replied:
leftover, this was used by me to run the action locally. Removing.

jmillan committed:
remove leftover

jmillan committed:
Update worker/src/RTC/SvcConsumer.cpp

jmillan committed:
Update worker/src/RTC/SvcConsumer.cpp

jmillan committed:
foo

jmillan committed:
clang-tidy: properly format fixes

jmillan committed:
const correctnes
